github/secret_scanning.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2022 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// SecretScanningService handles communication with the secret scanning related
+// methods of the GitHub API.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/secret-scanning
+type SecretScanningService service
+
+type SecretScanningAlert struct {
+	Number       *int64     `json:"number,omitempty"`
+	CreatedAt    *Timestamp `json:"created_at,omitempty"`
+	URL          *string    `json:"url,omitempty"`
+	HTMLURL      *string    `json:"html_url,omitempty"`
+	LocationsURL *string    `json:"locations_url,omitempty"`
+	State        *string    `json:"state,omitempty"`
+	Resolution   *string    `json:"resolution,omitempty"`
+	ResolvedAt   *Timestamp `json:"resolved_at,omitempty"`
+	ResolvedBy   *User      `json:"resolved_by,omitempty"`
+	SecretType   *string    `json:"secret_type,omitempty"`
+	Secret       *string    `json:"secret,omitempty"`
+}
+
+type SecretScanningAlertLocation struct {
+	Type    *string                             `json:"type,omitempty"`
+	Details *SecretScanningAlertLocationDetails `json:"details,omitempty"`
+}
+
+type SecretScanningAlertLocationDetails struct {
+	Path        *string `json:"path,omitempty"`
+	Startline   *int64  `json:"start_line,omitempty"`
+	EndLine     *int64  `json:"end_line,omitempty"`
+	StartColumn *int64  `json:"start_column,omitempty"`
+	EndColumn   *int64  `json:"end_column,omitempty"`
+	BlobSHA     *string `json:"blob_sha,omitempty"`
+	BlobURL     *string `json:"blob_url,omitempty"`
+	CommitSHA   *string `json:"commit_sha,omitempty"`
+	CommitURL   *string `json:"commit_url,omitempty"`
+}
+
+// SecretScanningAlertListOptions specifies optional parameters to the SecretScanningService.ListAlertsForEnterprise method.
+type SecretScanningAlertListOptions struct {
+	// State of the secret scanning alerts to list. Set to open or resolved to only list secret scanning alerts in a specific state.
+	State string `url:"state,omitempty"`
+
+	// A comma-separated list of secret types to return. By default all secret types are returned.
+	SecretType string `url:"secret_type,omitempty"`
+
+	// A comma-separated list of resolutions. Only secret scanning alerts with one of these resolutions are listed.
+	// Valid resolutions are false_positive, wont_fix, revoked, pattern_edited, pattern_deleted or used_in_tests.
+	Resolution string `url:"resolution,omitempty"`
+
+	ListCursorOptions
+}
+
+// SecretScanningAlertLocationListOptions specifies optional parameters to the SecretScanningService.ListLocationsForAlert method.
+type SecretScanningAlertLocationListOptions struct {
+	ListOptions
+}
+
+// SecretScanningAlertUpdateOptions specifies optional parameters to the SecretScanningService.UpdateAlert method.
```

Comment:
Let's please get rid of this and reuse the one above.

Reply:
Although these structs are very similar, the use cases and therefore the documentation is quite different.

Should I copy the documentation over to the `SecretScanningAlertListOptions`, making it clear the values have different meanings depending on where they're being used?

Reply:
Oh, OK... Then that is fine if you wish to leave these as-they-are.